collect_app/src/main/java/org/odk/collect/android/audio/AudioPlayerViewModel.java

```diff
@@ -19,10 +19,11 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Queue;
+import java.util.function.Supplier;
 
 class AudioPlayerViewModel extends ViewModel implements MediaPlayer.OnCompletionListener {
 
-    private final MediaPlayerFactory mediaPlayerFactory;
+    private final Supplier<MediaPlayer> mediaPlayerFactory;
```

Comment:
I ran into a problem with the `audio` stuff due to it instantiating a `Scheduler` so had to do a slight rework. I like the idea of using `Supplier` instead of an explicit `...Factory` or `...Provider` type now the new desugaring gives us access to it. We might still need a concrete class for Dagger but everything else should be able to use `Supplier`.